javagen/src/main/java/com/azure/autorest/Javagen.java

```diff
@@ -214,15 +214,15 @@ JavaPackage writeToTemplates(CodeModel codeModel, Client client, JavaSettings se
         }
 
         // Test
-        if (settings.isDataPlaneClient() && settings.isGenerateTests()) {
+        if (settings.isDataPlaneClient()) {
```

Comment:
I think `settings.isGenerateTests()` still needed?

I mean, default `generate-samples` and `generate-tests` to true in CadlPlugin or EmitterOptions.

User can still disable by explicitly setting it to false.

Reply:
You mean if user sets `generate-tests` to false, then the TestBase won't be generated as well? Makes sense, will do that.

Reply:
Yes, `TestBase` still follow the `generate-tests`. The change is when there is no tests, still generate the TestBase (so dev can add a new test class extends it, avoid writing these interceptorManager).